src/vcpkg/commands.add-version.cpp

```diff
@@ -393,15 +394,16 @@ namespace vcpkg::Commands::AddVersion
             msg::println_warning(msgAddVersionDetectLocalChangesError);
         }
 
-        for (auto&& port_name : port_names)
-        {
+        std::mutex mtx;
+
+        auto work = [&](const std::string& port_name) {
```

Comment:
@ras0219-msft The thing is that `try_load_port()` itself may call `Checks::check_exit()`. Given the fact that we sprinkle `Checks::check_exit` everywhere in our code, there is nearly nothing that we can parallelize!

IMO we should replace this by throwing an exception instead and print the error message properly by wrapping everything in `main()` inside a `try ... catch` and call `Checks::check_exit` from `catch`.

Reply:
Then as part of this, `try_load_port()` should be fixed to never call `Checks::check_exit()` and always return its errors via the existing return type of `Expected<>`.

Reply:
@ras0219-msft The formatting check can't be parallelized because it calls too many functions that may exit. Please also have a look at https://github.com/microsoft/vcpkg/issues/27152 .